### PR TITLE
Cleanup storehaus-testing

### PR DESCRIPTION
**Pull request by BenFradet**

### CODE REVIEW

**Comment by BenFradet** [comment]
Finally done!

File: storehaus-testing/src/main/scala/com/twitter/storehaus/testing/Cleanup.scala [hunk 1]
```diff
@@ -16,21 +16,28 @@
 
 package com.twitter.storehaus.testing
 
-import scala.collection.mutable.{ HashSet, SynchronizedSet }
+import java.util.Collections
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.collection.JavaConverters._
 import scala.util.control.Exception.allCatch
```
Comment (johnynek) [inline storehaus-testing/src/main/scala/com/twitter/storehaus/testing/Cleanup.scala]: allCatch predates `scala.util.Try`. Can we move to that?

File: storehaus-testing/src/main/scala/com/twitter/storehaus/testing/Cleanup.scala [hunk 2]
```diff
@@ -16,21 +16,28 @@
...
-  private val instances = new HashSet[Cleanup] with SynchronizedSet[Cleanup]
+  private val instances =
+    Collections.newSetFromMap(new ConcurrentHashMap[Cleanup, java.lang.Boolean]())
   /** Clean up all instances capturing ( and logging ) errors */
   def cleanup() {
-    instances.map(c => allCatch.either(c.cleanup()).left.map((c, _))).map(_.fold({
-      case (closeable, error) => println("failed to cleanup %s: %s".format(closeable, error.getMessage))
-    }, identity))
+    instances.asScala
+      .map(c => allCatch.either(c.cleanup()).left.map((c, _)))
```
Comment (johnynek) [inline storehaus-testing/src/main/scala/com/twitter/storehaus/testing/Cleanup.scala]:
``` scala
.foreach { c =>
  Try(c.cleanup()).recover {
    case error => println(s"failed to cleanup $c: ${error.getMessage}")
  }
}
```

**Review by johnynek — CHANGES_REQUESTED** [review]

**Comment by BenFradet** [comment]
@johnynek updated :+1: 
I chose to ignore mima regarding storehaus-testing, do you agree?

**Review by johnynek — APPROVED** [review]

**Comment by johnynek** [comment]
Thanks!

**Pull request merged**